### DISCUSSION TITLE
Trying to change text color-  on a toggle button.

### DISCUSSION THREAD

**Discussion by smcgann99** [HASwitchPlate/openHASP#566] [General]
The button is working and turning the gpio pin on and off as required.

I figured out that the background changes to the "Primary color" defined in HASP Design, when ON.
I assume that the text color is set to a default bright white ( I cannot see anywhere to change this ?)
I send a new color value when the button is on, as I wanted the text orange, but this seems to be ignored, I do see this color very briefly when I toggle it OFF again, just before it reverts to normal state.
There do seem to be a couple of errors after the button toggles, not sure if this blocks my new color ?

Prompt > [18:00:32.058][98292/110736 11][54220/54660  1] MQTT PUB: p1b3 => {"event":"down","val":0}
Prompt > [18:00:32.090][98292/109152  9][54220/54552  1] MQTT PUB: p1b3 => {"event":"up","val":1}
Prompt > [18:00:32.101][98292/109152  9][54220/54552  1] MQTT PUB: output20 => {"state":"on"}
Prompt > [18:00:32.152][102388/112236  8][54220/54764  1] MQTT RCV:  = {"event":"up","val":1}
Prompt > [18:00:32.163][98292/110980 11][54220/54764  1] ATTR: Unknown property event (60634)
Prompt > [18:00:32.174][98292/110980 11][54220/54764  1] ATTR: Unknown property val (15809)
Prompt > [18:00:32.261][102388/112268  8][54220/54764  1] MQTT RCV: p1b3.text_color = #dea031

**Comment by fvanroie** [comment]
You probably need to add the state suffix to the property: [Styling Properties - openHASP](https://www.openhasp.com/0.7.0/design/styling/#states)
e.g. p1b3.text_color01
01 = styling for toggled state
